pkg/storage/sqlite/sqlite.go

```diff
@@ -92,21 +97,104 @@ func (s SQLite) CreateMovie(ctx context.Context, movie model.Movie) (int64, erro
 	if movie.ID != 0 {
 		insertColumns = table.Movie.AllColumns
 	}
-	stmt := table.Movie.INSERT(insertColumns).RETURNING(table.Movie.ID).MODEL(movie).ON_CONFLICT(table.Movie.ID).DO_UPDATE(sqlite.SET(
-		table.Movie.MutableColumns.SET(sqlite.ROW(setColumns...)),
-	))
-	result, err := s.handleInsert(ctx, stmt)
+
+	stmt := table.Movie.
+		INSERT(insertColumns).
+		MODEL(movie.Movie).
+		RETURNING(table.Movie.ID).
+		ON_CONFLICT(table.Movie.ID).
+		DO_UPDATE(sqlite.SET(table.Movie.MutableColumns.SET(sqlite.ROW(setColumns...))))
+
+	result, err := stmt.ExecContext(ctx, tx)
 	if err != nil {
 		return 0, err
 	}
 
 	inserted, err := result.LastInsertId()
 	if err != nil {
+		tx.Rollback()
+		return 0, err
+	}
+
+	state := storage.MovieTransition{
+		MovieID:    int32(inserted),
+		ToState:    string(storage.MovieStateMissing),
+		MostRecent: true,
+		SortKey:    1,
```

Comment:
How does the sort key work? Is it meant to increment

Reply:
Oh it's sorting the transitions per movie?

Reply:
yes, it increments on each transition change